blocks/library/embed/index.js

```diff
@@ -200,18 +200,16 @@ function getEmbedBlockSettings( { title, icon, category = 'embed', transforms, k
 									html={ html }
 									title={ iframeTitle }
 									type={ type }
-									onFocus={ () => setFocus() }
 								/>
 							</div>
 						) }
-						{ ( caption && caption.length > 0 ) || !! focus ? (
+						{ ( caption && caption.length > 0 ) || isSelected ? (
 							<RichText
 								tagName="figcaption"
 								placeholder={ __( 'Write caption…' ) }
 								value={ caption }
-								focus={ focus }
-								onFocus={ setFocus }
```

Comment:
This change regressed on the fix introduced by #4011, where in that case I explicitly wanted to handle focus within the iframe. Can we otherwise expose manually setting a block as selected?

Reply:
You mean like a `onSelect` prop or something? I'd prefer to avoid it in favor of a global event but if it's not possible for iframes, we could provide it

Reply:
We could see about generalizing the `blur` check which occurs in Sandbox, to see if focus has transitioned to an iframe within a block:

https://github.com/WordPress/gutenberg/blob/98d6c7e538c3cdc22e11ee4333d541f82125be20/components/sandbox/index.js#L54-L73